nsot/util/core.py

```diff
@@ -116,6 +116,35 @@ def slugify(s):
     return s
 
 
+def slugify_interface(interface=None, device_hostname=None, name=None,
```

Comment:
Based on a Slack discussion we're going to come up w/ a generic model-aware mapping of input data to natural key "soon". This is fine for now.